src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.hpp

```diff
@@ -73,23 +73,28 @@ class MGRStrategyBase
 {
 public:
 
-  static constexpr HYPRE_Int numLevels = NLEVEL;       ///< Number of levels
+  static constexpr HYPRE_Int numLevels = NLEVEL;              ///< Number of levels
 
 protected:
 
-  HYPRE_Int m_numBlocks{ 0 };                          ///< Number of different matrix blocks treated separately
+  HYPRE_Int m_numBlocks{ 0 };                                 ///< Number of different matrix blocks treated separately
 
-  std::vector< HYPRE_Int > m_labels[numLevels]{};      ///< Dof labels kept at each level
-  HYPRE_Int m_numLabels[numLevels]{ -1 };              ///< Number of dof labels kept
-  HYPRE_Int * m_ptrLabels[numLevels]{ nullptr };       ///< Pointers to each level's labels, as consumed by MGR
+  std::vector< HYPRE_Int > m_labels[numLevels]{};             ///< Dof labels kept at each level
+  HYPRE_Int m_numLabels[numLevels]{ -1 };                     ///< Number of dof labels kept
+  HYPRE_Int * m_ptrLabels[numLevels]{ nullptr };              ///< Pointers to each level's labels, as consumed by MGR
 
-  MGRFRelaxationType m_levelFRelaxType[numLevels];           ///< F-relaxation type for each level
-  HYPRE_Int m_levelFRelaxIters[numLevels]{ -1 };             ///< Number of F-relaxation iterations for each level
-  MGRInterpolationType m_levelInterpType[numLevels];         ///< Interpolation type for each level
-  MGRRestrictionType m_levelRestrictType[numLevels];         ///< Restriction type for each level
-  MGRCoarseGridMethod m_levelCoarseGridMethod[numLevels];    ///< Coarse grid method for each level
+  MGRFRelaxationType m_levelFRelaxType[numLevels];            ///< F-relaxation type for each level
+  HYPRE_Int m_levelFRelaxIters[numLevels]{ -1 };              ///< Number of F-relaxation iterations for each level
+  MGRInterpolationType m_levelInterpType[numLevels];          ///< Interpolation type for each level
+  MGRRestrictionType m_levelRestrictType[numLevels];          ///< Restriction type for each level
+  MGRCoarseGridMethod m_levelCoarseGridMethod[numLevels];     ///< Coarse grid method for each level
   MGRGlobalSmootherType m_levelGlobalSmootherType[numLevels]; ///< Global smoother type for each level
-  HYPRE_Int m_levelGlobalSmootherIters[numLevels]{ -1 };     ///< Number of global smoother iterations for each level
+  HYPRE_Int m_levelGlobalSmootherIters[numLevels]{ -1 };      ///< Number of global smoother iterations for each level
+#if GEOS_USE_HYPRE_DEVICE == GEOS_USE_HYPRE_CPU
+  HYPRE_Real m_coarseGridThreshold{ 1.0e-20 };                ///< Coarse grid truncation threshold
+#else
+  HYPRE_Real m_coarseGridThreshold{ 0.0 };                    ///< Coarse grid truncation threshold
+#endif
```

Comment:
this doesn't make sense to me? I read it as 
"If we `GEOS_USE_HYPRE_DEVICE` is the same as `GEOS_USE_HYPRE_CPU` then do something"...but should these ever be the same?

Reply:
I've seen a few test cases with GPUs where `threshold > 0` causes a crash in hypre, which does not seem straightforward to resolve. Since `threshold > 0` works for CPUs, and for the sake of getting this new truncation functionality used in GEOS, I'm currently setting different threshold according to the architecture. After the GPU path in hypre is improved, I will change back the `threshold` parameter to the same default for CPUs and GPUs in GEOS